### PR TITLE
Ipam vlan readme

### PR DESCRIPTION
**Pull request by henderiw**
This is a PR to update the readme for IPAM/VLAN fn

### CODE REVIEW

File: krm-functions/vlan-fn/README.md [hunk 1]
```diff
@@ -1,23 +1,7 @@
...
-- default network:
-    - No child resources are created since the interface is attached to the default network of the k8s cluster and all requirements will be satisfied within the k8s cluster. Its status will be determined as True, since there are no dependencies.
-- No CNI type present:
-    - When no CNI type is present this is seen as a loopback interface request for which only an IPAllocation (kind loopback) child resource will be requested
-- CNI Type present:
-    - When a CNI type is present the CNI Type of the interface request is validated against the cluster. If no match is found an error is returned
-    - If the CNI type matches the cluster context a NAD, IPAllocation (kind network) and potentially a VLANAllocation is requested based on the content of the attachementType in.
-
-Only when all child/`own` resources are satisfied the status is determined as True. The interface-fn will update the status in its Status field of the Interface KRM resource.
+The ipam-fn is a KRM function leveraging the `cond sdk` using the vlan.alloc.nephio.org/v1alpha1.VLANAllocation as a `for` KRM resource.
```
Comment (efiacor) [inline krm-functions/vlan-fn/README.md:3]: vlan-fn?

File: krm-functions/ipam-fn/README.md [hunk 2]
```diff
@@ -1,23 +1,7 @@
...
-- No CNI type present:
-    - When no CNI type is present this is seen as a loopback interface request for which only an IPAllocation (kind loopback) child resource will be requested
-- CNI Type present:
-    - When a CNI type is present the CNI Type of the interface request is validated against the cluster. If no match is found an error is returned
-    - If the CNI type matches the cluster context a NAD, IPAllocation (kind network) and potentially a VLANAllocation is requested based on the content of the attachementType in.
-
-Only when all child/`own` resources are satisfied the status is determined as True. The interface-fn will update the status in its Status field of the Interface KRM resource.
+The ipam-fn is a KRM function leveraging the `cond sdk` using the ipam.alloc.nephio.org/v1alpha1.IPAllocation as a `for` KRM resource.
+The function allocates IP from an IPAM backend based on the content of the IPAllocation. A function is implemented to align with the fn sdk, but more importantly it allows us to use the fn in a `kpt` pipeline w/o relying on porch. When used in the `kpt` pipeline it uses a stub backend for testing purposes
```
Comment (efiacor) [inline krm-functions/ipam-fn/README.md:4]: Is this necessary to mention?
Should it be expanded upon in a separate section?

**Review by efiacor — COMMENTED** [review]
Few minor comments

File: krm-functions/ipam-fn/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (liamfallon) [reply krm-functions/ipam-fn/README.md:4]: This rephrasing incorporates the comments above:

The ipam-fn is a KRM function leveraging the `cond sdk`. It uses the ipam.alloc.nephio.org/v1alpha1.IPAllocation as a `for` KRM resource.

The function allocates IP addresses from an IPAM backend based on the content of the IPAllocation. The function is implemented to align with the `fn sdk` but, more importantly, the function can be used in a `kpt` pipeline without relying on porch. When used in a `kpt` pipeline, a stub backend can be deployed for testing purposes.

File: krm-functions/vlan-fn/README.md (hunk 1 above)
Comment (liamfallon) [reply krm-functions/vlan-fn/README.md:3]: Maybe rephrase as:

The vlan-fn is a KRM function leveraging the `cond sdk`. It uses the vlan.alloc.nephio.org/v1alpha1.VLANAllocation as a `for` KRM resource.

The function allocates VLANs from a VLAN backend based on the content of the VLANAllocation. The function is implemented to align with the `fn sdk` but, more importantly, the function can be used in a `kpt` pipeline without relying on porch. When used in a `kpt` pipeline, a stub backend can be deployed for testing purposes.

Comment (henderiw) [reply krm-functions/vlan-fn/README.md]: thx, appreciate.  I updated the text based on your feedback.

File: krm-functions/ipam-fn/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (henderiw) [reply krm-functions/ipam-fn/README.md]: thx updated the text
